src/SmartFormat/SmartFormatter.cs

```diff
@@ -201,8 +252,21 @@ public string FormatWithCache(ref FormatCache? cache, string format, params obje
         /// <param name="args">The objects to format.</param>
         public void FormatWithCacheInto(ref FormatCache cache, IOutput output, string format, params object[] args)
         {
+            FormatWithCacheInto(ref cache, output, args, format);
+        }
+
+        /// <summary>
+        /// Writes the formatting result into an <see cref="IOutput"/> instance, using the <see cref="FormatCache"/>.
+        /// </summary>
+        /// <param name="cache">The <see cref="FormatCache"/> to use.</param>
+        /// <param name="output">The <see cref="IOutput"/> where the result is written to.</param>
+        /// <param name="args">The objects to format.</param>
+        /// <param name="format">The format string.</param>
+        public void FormatWithCacheInto(ref FormatCache cache, IOutput output, IList<object> args, string format)
+        {
+            args = args ?? k_Empty;
```

Comment:
Should read:

```args ??= k_Empty;```

in case args is `nullable` - otherwise this line is redundant